src/arrayMethodReduce.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result;
+    let start;
+
+    if (arguments.length === 2) {
+      result = initialValue;
+      start = 0;
+    } else {
+      result = this[0];
+      start = 1;
+    }
+
+    for (let i = start; i < this.length; i++) {
+      result = callback(result, this[i], i, this);
+    }
```

Comment:
We on `Array iteration methods` topic. So maybe should use some array method instead of a simple loop )